.github/workflows/terraform.yml

```diff
@@ -0,0 +1,38 @@
+name: Terraform
+
+on:
+  push:
+    branches:
+      - main
+      - terraform-sharing-db
+  pull_request:
+    branches:
+      - main
+      - terraform-sharing-db
+
+jobs:
+  terraform:
+    name: 'Terraform'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 1.0.0
+
+      - name: Terraform Init
+        run: terraform init
+        working-directory: ./terraform/state
```

Comment:
this should be the postgres directory. state, networking, and iam provider/role have already been set up.